openslides_backend/action/action.py

```diff
@@ -1,139 +1,408 @@
-        for name, action in actions_map.items():
-            if getattr(action, "is_dummy", False):
-                yield name, "Not implemented"
-            else:
-                yield name, action.schema
+        self.user_id = user_id
+        self.check_permissions(payload)
+        self.validate(payload)
+
+        instances = self.get_updated_instances(payload)
+        for instance in instances:
+            self.validate_instance(instance)
+            self.base_update_instance(instance)
 
-    def handle_request(self, payload: Payload, user_id: int) -> ActionResponse:
+            self.write_request_elements.extend(
+                self.handle_relation_updates(model, instance)
+            )
+            self.write_request_elements.extend(
+                self.create_write_request_elements(instance)
+            )
+        yield from self.process_write_request_elements()
+
+    def check_permissions(self, payload: ActionPayload) -> None:
         """
-        Takes payload and user id and handles this request by validating and
-        parsing all actions. In the end it sends everything to the event store.
+        Checks permission by requesting permission service.
         """
-        self.user_id = user_id
+        try:
+            self.additions = self.permission.is_allowed(
+                self.name, self.user_id, list(payload)
+            )
+        except NotAllowed as e:
+            raise PermissionDenied(
+                f"You are not allowed to perform action {self.name}: {e.reason}"
+            )
+
+    def get_updated_instances(self, payload: ActionPayload) -> ActionPayload:
+        """
+        By default this does nothing. Override in subclasses to adjust the updates
+        to all instances of the payload. You can only update instances of the model
+        of this action.
+        If needed, this can also be used to do additional validation on the whole
+        payload.
+        """
+        yield from payload
 
-        # Validate payload of request
+    def validate_instance(self, payload: ActionPayload) -> None:
+        """
+        Validates action payload according to schema class attribute.
+        """
         try:
-            self.validate(payload)
+            type(self).schema_validator(payload)
         except fastjsonschema.JsonSchemaException as exception:
             raise ActionException(exception.message)
 
-        # TODO: Start a loop here and retry parsing actions and writing to event
-        # store for some time if event store sends ModelLocked Exception
+    def base_update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Updates one instance of the payload. This can be overridden by custom
+        action classes.
+        """
+        return instance
 
-        # Parse actions and creates events
-        write_request_element, results = self.parse_actions(payload)
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
I think this method is not useful here. Where do we need it? The code can also be in base_update_instance and if some mixins want to create generic code they can define `special_update_instance` method.

Reply:
Yes, I generally agree, but all three generic actions (create, update, delete) use such a method and it would therefore be defined in all three files. I found that unnecessary and moved it one level up. We could write `return self.update_instance(instance)` in the `base_update_instance` to symbolize the usage, if you want. Or make a better suggestion, I'm not so sure about this either.

Reply:
Ok, I follow your suggestion in #358